pkg/solana/config/chain_reader.go

```diff
@@ -8,94 +8,85 @@ import (
 	"github.com/gagliardetto/solana-go/rpc"
 
 	commoncodec "github.com/smartcontractkit/chainlink-common/pkg/codec"
-	"github.com/smartcontractkit/chainlink-common/pkg/codec/encodings"
-	"github.com/smartcontractkit/chainlink-common/pkg/codec/encodings/binary"
-	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
 )
 
-type ChainReader struct {
-	Namespaces map[string]ChainReaderMethods `json:"namespaces" toml:"namespaces"`
+type ContractReader struct {
+	Namespaces map[string]ChainContractReader `json:"namespaces"`
 }
 
-type ChainReaderMethods struct {
-	Methods map[string]ChainDataReader `json:"methods" toml:"methods"`
+type ChainContractReader struct {
+	codec.IDL `json:"anchorIDL"`
+	// Reads key is the off-chain name for this read.
+	Reads map[string]ReadDefinition `json:"reads"`
+	// TODO ContractPollingFilter same as EVM?
 }
 
-type ChainDataReader struct {
-	AnchorIDL string `json:"anchorIDL" toml:"anchorIDL"`
-	// Encoding defines the type of encoding used for on-chain data. Currently supported
-	// are 'borsh' and 'bincode'.
-	Encoding  EncodingType         `json:"encoding" toml:"encoding"`
-	Procedure ChainReaderProcedure `json:"procedure" toml:"procedure"`
+type ReadDefinition struct {
+	ChainSpecificName   string                      `json:"chainSpecificName"`
+	ReadType            ReadType                    `json:"readType,omitempty"`
+	InputModifications  commoncodec.ModifiersConfig `json:"inputModifications,omitempty"`
+	OutputModifications commoncodec.ModifiersConfig `json:"outputModifications,omitempty"`
+	RPCOpts             *RPCOpts                    `json:"rpcOpts,omitempty"`
```

Comment:
Are the `RPCOpts` necessary anymore? We are explicitly using finalized commitment and encoding will probably always be `base64`. The only remaining option is for data slices. Do we use that or are we expecting to use that?

Reply:
Idk, I haven't looked into this yet, just kept it because it was already there

Reply:
it can most likely be removed